### ISSUE TITLE
Making 'enter' easier

### ISSUE THREAD

**Issue by kuanxu** [JunoLab/Juno.jl#113]
Currently, to ‘enter’ a line to begin debugging, one has to 

1. run the file once first,
2. then add the `@enter` macro manually at the beginning of the line,
3. If this is a line where value assignment is involved, one also has to remove the variable on the left-hand side as well as the equal sign. 

Any of these can significantly drag the speed of the debugging work, let alone the combination, and it is very very error-prone as one has to modify the code unnecessarily. So it'll be great that debugging can start as simply as in Matlab - one can simply start debugging from the first line of the code or the line where the cursor is by just clicking __one__ button. In GDB, this can be done by `run` (if I recall correctly). 

Of course, getting rid of the three steps above is beyond just the Juno wrapper for the debugger, but the design of the debugger too. So I'm filing a ticket too [here](https://github.com/Keno/ASTInterpreter2.jl/issues/19).

**Comment by pfitzseb** [comment]
One thing that could be implemented on the Juno side: We could allow selecting a function call and adding a "Step into selected function call" in the context menu (and as a command, obviously -- maybe even a toolbar button), which would then call `@enter selected_function_call(...)` internally.

Do you think that is sufficiently convenient? ;)

**Comment by pfitzseb** [comment]
Actually, now that I'm thinking about this -- we could even have a small toolbar pop up whenever you select a valid function call and offer some options about what you could do with it: Profile it, step into it, benchmark it etc.

**Comment by kuanxu** [comment]
@pfitzseb That's a neat idea!!

In the meanwhile, I still wish that I can start to traverse the entire file from the very beginning, starting by a simple click. People do this when trying to go through a file to understand the working/logical flow. How do you think?

**Comment by pfitzseb** [comment]
Are you talking about stepping through a "top-level" file? If so then you could just use inline evaluation (with `Ctrl-Shift-Enter`) to quickly step through it.

**Comment by kuanxu** [comment]
Yes and no. I meant stepping through a "top-level" file but being able to step into lower levels whenever I want to. 

I just tried `Ctrl-Shift-Enter` but nothing happened. I must have missed something. Can you elaborate a little?

**Comment by pfitzseb** [comment]
I see, makes sense. I don't think we can easily do that with the debugger though -- my proposal above is probably the best thing possible for now.

`Ctrl-Shift-Enter` should be `Shift-Enter` above, which evaluates the current block and steps to the next one.

**Comment by kuanxu** [comment]
`Shift-Enter` works. But it does invoke the debugger. I can only see the function which is to be called next. I wish I could, at that point, press F11 to get into the called function.

**Comment by daravi** [comment]
It would be great if there was a Start Debugging option that would just start at the top of the active file.

It could then be configured (in preference) to stop at entry (like enter) or run until the first breakpoint (like run). We could also add Debug Block to parallel Run Block if there is demand for it.
